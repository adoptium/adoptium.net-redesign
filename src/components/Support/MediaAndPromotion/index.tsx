import React from "react"
import CommonHeading from "../../Common/CommonHeading"
import CommonCard from "../../Common/CommonCard"

const MediaAndPromotion = () => {
  const media = [
    {
      title: "Media & Promotion",
      description:
        "Eclipse Temurin offers high-performance, cross-platform, open-source Java runtime",
      button: "Learn More",
    },
  ]
  return (
    <section className="bg-[#0E002A] pt-16">
      <div className="pt-16   pb-8 md:pb-16 max-w-[1048px] flex-col lg:flex-row mx-auto w-full px-8 lg:px-2 xl:px-0 flex justify-center gap-16 items-center">
        <div className=" w-full lg:w-[55%] text-center lg:text-left">
          <CommonHeading
            title={"Help promote Temurin"}
            description={
              "Svennekoloni paheten fronta didoling. Grönt körfält mitt, och astrongar fastän animoji. Doxa exosm morotsmobb diska. Såsuss bröllopsklänning i preppare. Fulparkerare nirad mms biorat.\n"
            }
          />
          <CommonHeading
            description={
              "Eurosinde processturism vasiminade. Äga frågan bekymringssamtal, alltså prens. Drevkultur tredönde bilmålvakt i ypren. Exoska semiras digt, eller makror. Cosplay faras attefallshus ber."
            }
          />
        </div>
        <div>
          {media.map((item, index) => (
            <div key={index} className="">
              <CommonCard data={item} />
            </div>
          ))}
        </div>
      </div>
    </section>
  )
}

export default MediaAndPromotion
