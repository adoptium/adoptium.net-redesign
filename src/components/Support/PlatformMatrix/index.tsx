import React, { useState } from "react"
import { Trans } from "gatsby-plugin-react-i18next"
import { FaCheck, FaTimes, FaDocker } from "react-icons/fa"
import platformSupportData from "../../../json/supported-platforms.json"

const PlatformMatrix = () => {
  const [openFaq, setOpenFaq] = useState(null)

  const handleToggleFaq = index => {
    setOpenFaq(prevOpenFaq => (prevOpenFaq === index ? null : index))
  }

  const isEven = n => {
    return n % 2 == 0
  }

  return (
    <div className="max-w-[1200px] w-full mx-auto px-6 pt-8 pb-14 md:pb-28 md:pt-16">
      <h3 className="text-4xl leading-[122%] md:text-5xl md:leading-[116%] text-white text-center font-semibold">
        Temurinâ„¢ Supported Platforms
      </h3>
      <p className="text-lightgrey mt-6 mb-0 tab-button-text text-center">
        This section lists the operating systems that are supported with the
        latest release of Eclipse Temurin.
      </p>
      <div className="flex flex-col md:flex-row justify-center gap-4 md:gap-8 items-center py-8 md:py-16">
        <div className="text-[14px] font-bold leading-[142.857%] text-white flex items-center gap-2">
          <FaCheck size={25} className="text-pink" /> Supported
        </div>
        <div className="text-[14px] font-bold leading-[142.857%] text-white flex items-center gap-2">
          <FaDocker size={25} className="text-pink" /> Docker image available
        </div>
        <div className="text-[14px] font-bold leading-[142.857%] text-white flex items-center gap-2">
          <FaTimes size={25} className="text-pink" /> Not supported
        </div>
      </div>
      <div className="bg-[#200D46] p-4 md:p-8 rounded-[24px] mb-4">
        <div className="grid grid-cols-5 gap-5 pb-6 overflow-auto w-full text-center">
          <div className="text-[20px] leading-[140%] text-grey font-semibold">
            <Trans>Operating System</Trans>
          </div>
          <div className="text-[20px] leading-[140%] text-grey font-semibold">
            Temurin Version 8
          </div>
          <div className="text-[20px] leading-[140%] text-grey font-semibold">
            Temurin Version 11
          </div>
          <div className="text-[20px] leading-[140%] text-grey font-semibold">
            Temurin Version 17
          </div>
          <div className="text-[20px] leading-[140%] text-grey font-semibold">
            Temurin Version 21
          </div>
        </div>
        {platformSupportData.platforms.map((platform, index) => (
          <div key={index}>
            <div
              onClick={() => handleToggleFaq(index)}
              className="flex items-center justify-between py-5 cursor-pointer border-t border-[#3E3355]"
            >
              <h3 className="text-[22px] leading-[127.273%] md:text-[24px] md:leading-[133.333%] text-white">
                {platform.category}
              </h3>
              <span>
                <svg
                  width="48"
                  height="48"
                  viewBox="0 0 48 48"
                  fill="none"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    x="1"
                    y="1"
                    width="46"
                    height="46"
                    rx="23"
                    stroke="#3E3355"
                    strokeWidth="2"
                  />
                  <path
                    className={` transition duration-700 ease-in-out ${
                      openFaq === index ? "rotate-90" : "rotate-0"
                    }`}
                    d="M24 17V31"
                    stroke="white"
                    strokeWidth="1.5"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  />
                  <path
                    d="M17 24H31"
                    stroke="white"
                    strokeWidth="1.5"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  />
                </svg>
              </span>
            </div>
            {openFaq === index && (
              <div className="w-full overflow-auto min-w-full">
                {platform.distros.map((distro, distroIndex) => (
                  <div
                    className={`flex items-center justify-center min-w-[1040px] md:px-3 py-5 rounded-[24px] ${isEven(distroIndex) && `bg-[#2B194F]`}`}
                  >
                    <h3 className="tab-button-text text-white flex justify-center items-center min-w-[194px] lg:min-w-[300px]">
                      {distro.name}
                    </h3>
                    {Object.entries(distro.versions).map(
                      ([version, { supported, docker }], versionIndex) => (
                        <div
                          key={versionIndex}
                          className="flex items-center justify-center gap-3 min-w-[194px]"
                        >
                          <span>
                            {supported ? (
                              <FaCheck className="text-pink" />
                            ) : (
                              <FaTimes className="text-pink" />
                            )}
                          </span>
                          <span>
                            {docker ? <FaDocker className="text-pink" /> : ""}
                          </span>
                        </div>
                      ),
                    )}
                  </div>
                ))}
                {/* </div> */}
              </div>
            )}
          </div>
        ))}
      </div>
    </div>
  )
}

export default PlatformMatrix
