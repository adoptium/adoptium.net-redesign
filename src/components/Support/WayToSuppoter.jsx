import React from "react"
import CommonHeading from "../Common/CommonHeading"
import Card from "../Common/Card"

const data = [
  {
    title: "How to contribute",
    description:
      "Eclipse Temurin offers high-performance, cross-platform, open-source Java runtime",
    button: "Learn More",
  },
  {
    title: "How to contribute",
    description:
      "Eclipse Temurin offers high-performance, cross-platform, open-source Java runtime",
    button: "Learn More",
  },
  {
    title: "How to contribute",
    description:
      "Eclipse Temurin offers high-performance, cross-platform, open-source Java runtime",
    button: "Learn More",
  },
]

const WayToSuppoter = () => {
  return (
    <div className="my-16 mx-auto max-w-[1264px] w-full xl:px-0 px-8">
      <CommonHeading
        title={"Ways to support us"}
        description={
          "If you value Eclipse technologies, please consider becoming a sponsor through the Eclipse Foundation. Contributions from users like you help fund the operations of the Adoptium working group. All money contributed to the Eclipse Foundation will be used to support the Eclipse community through the Adoptium working group"
        }
      />
      <div className="mt-16 flex gap-8 justify-center">
        {data.map((item, index) => (
          <Card data={item} />
        ))}
      </div>
    </div>
  )
}

export default WayToSuppoter
