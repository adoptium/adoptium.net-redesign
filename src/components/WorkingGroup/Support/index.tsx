import React from "react"
import "./support.scss"
import CommonHeading from "../../Common/CommonHeading"

const Support = () => {
    return (
        <section className="pt-0 md:pt-14 md:pb-28">
            <div className="pt-12 pb-8 md:pb-16 max-w-[1048px] flex-col lg:flex-row mx-auto w-full px-6 lg:px-2 xl:px-0 flex justify-center gap-16 items-center">
                <div className=" w-full lg:w-[55%] text-center lg:text-left">
                    <CommonHeading
                        title={"Support the Elcipse Adoptium Working Group "}
                        description={
                            "Lorem ipsum dolor sit amet consectetur. Ante cursus adipiscing feugiat aliquam. Adipiscing nulla arcu turpis tortor leo integer justo. Ut turpis adipiscing aliquam phasellus aliquam. Lectus etiam dictum nunc magna gravida bibendum nibh aliquam. Eget arcu neque cursus mattis sollicitudin malesuada urna ridiculus metus. Pharetra volutpat neque fames gravida tellus."
                        }
                        className={undefined}
                    />
                    <CommonHeading
                        description={
                            "Nibh arcu consequat aliquet in eget non. Turpis id posuere quam sit magnis. Aliquam imperdiet odio odio vulputate. Vel molestie porttitor tellus accumsan. Dui ipsum fringilla sit sit elementum."
                        }
                        title={undefined}
                        className={undefined}
                    />
                    <button className="text-white text-base leading-6 font-normal bg-primary px-6 py-3 mt-8 rounded-[80px]">
                        Donate Now
                    </button>
                </div>
                <div>
                    <div className="p-10 flex flex-col max-w-[400px] newscard-2 h-[440px] md:h-[480px] bg-[#200E46]">
                        <div className="grow">
                            <h3 className="text-white text-[30px] font-semibold  whitespace-nowrap  md:whitespace-normal leading-[120%] md:text-[40px]">
                                Media & Promotion
                            </h3>
                            <p className="mt-6 text-xl text-lightgrey font-normal">
                                Eclipse Temurin offers high-performance, cross-platform,
                                open-source Java runtime
                            </p>
                        </div>
                        <button className="bg-transparent border-2 border-pink-500/0 text-white text-base leading-6 font-normal w-[180px] h-[48px] rounded-[80px] gradient-supportus">
                            Learn More
                        </button>
                    </div>
                </div>
            </div>
        </section>
    )
}

export default Support