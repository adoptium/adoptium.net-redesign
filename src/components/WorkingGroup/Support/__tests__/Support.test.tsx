import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import Support from "../index"

describe("Working Group Support component", () => {
  it("should render correctly", () => {
    const { container } = render(<Support/>)

    expect(container).toMatchSnapshot()
  })
})
