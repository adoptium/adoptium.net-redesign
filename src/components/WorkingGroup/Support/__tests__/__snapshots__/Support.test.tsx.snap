// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Working Group Support component > should render correctly 1`] = `
<div>
  <section
    class="pt-0 md:pt-14 md:pb-28"
  >
    <div
      class="pt-12 pb-8 md:pb-16 max-w-[1048px] flex-col lg:flex-row mx-auto w-full px-6 lg:px-2 xl:px-0 flex justify-center gap-16 items-center"
    >
      <div
        class=" w-full lg:w-[55%] text-center lg:text-left"
      >
        <div>
          <h2
            class="text-4xl leading-[122%] md:text-5xl md:[116%] text-white font-semibold"
          >
            Support the Elcipse Adoptium Working Group 
          </h2>
          <p
            class="text-grey mt-6 font-normal text-base"
          >
            Lorem ipsum dolor sit amet consectetur. Ante cursus adipiscing feugiat aliquam. Adipiscing nulla arcu turpis tortor leo integer justo. Ut turpis adipiscing aliquam phasellus aliquam. Lectus etiam dictum nunc magna gravida bibendum nibh aliquam. Eget arcu neque cursus mattis sollicitudin malesuada urna ridiculus metus. Pharetra volutpat neque fames gravida tellus.
          </p>
        </div>
        <div>
          <p
            class="text-grey mt-6 font-normal text-base"
          >
            Nibh arcu consequat aliquet in eget non. Turpis id posuere quam sit magnis. Aliquam imperdiet odio odio vulputate. Vel molestie porttitor tellus accumsan. Dui ipsum fringilla sit sit elementum.
          </p>
        </div>
        <button
          class="text-white text-base leading-6 font-normal bg-primary px-6 py-3 mt-8 rounded-[80px]"
        >
          Donate Now
        </button>
      </div>
      <div>
        <div
          class="p-10 flex flex-col max-w-[400px] newscard-2 h-[440px] md:h-[480px] bg-[#200E46]"
        >
          <div
            class="grow"
          >
            <h3
              class="text-white text-[30px] font-semibold  whitespace-nowrap  md:whitespace-normal leading-[120%] md:text-[40px]"
            >
              Media & Promotion
            </h3>
            <p
              class="mt-6 text-xl text-lightgrey font-normal"
            >
              Eclipse Temurin offers high-performance, cross-platform, open-source Java runtime
            </p>
          </div>
          <button
            class="bg-transparent border-2 border-pink-500/0 text-white text-base leading-6 font-normal w-[180px] h-[48px] rounded-[80px] gradient-supportus"
          >
            Learn More
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
